addons/stock_landed_costs/models/stock_landed_costs.py

```diff
@@ -0,0 +1,273 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import product
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.tools import float_compare, float_round
+import odoo.addons.decimal_precision as dp
+
+
+class StockLandedCost(models.Model):
+    _name = 'stock.landed.cost'
+    _description = 'Stock Landed Cost'
+    _inherit = 'mail.thread'
+
+    name = fields.Char(track_visibility='always', default=lambda self: self.env['ir.sequence'].next_by_code('stock.landed.cost'), readonly=True, copy=False)
+    date = fields.Date(required=True, states={'done': [('readonly', True)]}, default=fields.Date.context_today, track_visibility='onchange', copy=False)
+    picking_ids = fields.Many2many('stock.picking', string='Pickings', states={'done': [('readonly', True)]}, copy=False)
+    cost_line_ids = fields.One2many('stock.landed.cost.lines', 'cost_id', string='Cost Lines', states={'done': [('readonly', True)]})
+    valuation_adjustment_line_ids = fields.One2many('stock.valuation.adjustment.lines', 'cost_id', string='Valuation Adjustments', states={'done': [('readonly', True)]})
+    description = fields.Text(string='Item Description', states={'done': [('readonly', True)]})
+    amount_total = fields.Float(compute='_compute_amount_total', string='Total', digits=dp.get_precision('Account'), store=True,
+                                track_visibility='always')
+    state = fields.Selection([('draft', 'Draft'), ('done', 'Posted'), ('cancel', 'Cancelled')], default='draft', readonly=True, track_visibility='onchange', copy=False)
+    account_move_id = fields.Many2one('account.move', string='Journal Entry', readonly=True, copy=False)
+    account_journal_id = fields.Many2one('account.journal', string='Account Journal', required=True)
+
+    @api.depends('cost_line_ids', 'cost_line_ids.price_unit', 'cost_line_ids.cost_id')
+    def _compute_amount_total(self):
+        for landed_cost_line in self:
+            landed_cost_line.amount_total = sum(landed_cost_line.cost_line_ids.mapped('price_unit'))
+
+    def get_valuation_lines(self):
```

Comment:
By seeing old function definition:
def get_valuation_lines(self, cr, uid, ids, picking_ids=None, context=None):
It seems something went wrong here.